tests/unittests/library/test_nsx_common.py

```diff
@@ -628,6 +637,151 @@ def test_nsx_read(self):
             'c'
         )
 
+    @pytest.mark.internal
+    @pytest.mark.unit
+    def test_nsx_search(self):
+        """Check nsx_common.nsx_search func"""
+        self._regen_ctx()
+        client_session = mock.Mock()
+
+        # we raise error?
+        client_session.read = mock.Mock(return_value={
+            'body': {}, 'status': 300
+        })
+        with self.assertRaises(cfy_exc.NonRecoverableError):
+            common.nsx_search(client_session, 'body/a', 'b', 'secret',
+                              uri_parameters={'scopeId': 'scopeId'})
+        # no results
+        client_session.read = mock.Mock(return_value={
+            'body': {}, 'status': 204
+        })
+        self.assertEqual(
+            common.nsx_search(client_session, 'body/a', 'a', 'secret',
+                              uri_parameters={'scopeId': 'scopeId'}),
+            (None, None)
+        )
+
+        # dict result
+        client_session.read = mock.Mock(return_value={
+            'body': {'a': {'name': 'b', 'objectId': 'c'}}, 'status': 204
+        })
+        self.assertEqual(
+            common.nsx_search(client_session, 'body/a', 'b', 'secret',
+                              uri_parameters={'scopeId': 'scopeId'}),
+            ('c', {'name': 'b', 'objectId': 'c'})
+        )
+
+        # list result
+        client_session.read = mock.Mock(return_value={
+            'body': {'a': [{
+                'name': 'b',
+                'objectId': 'c'
+            }, {
+                'name': 'd',
+                'objectId': 'e'
+            }]}, 'status': 204
+        })
+        self.assertEqual(
+            common.nsx_search(client_session, 'body/a', 'd', 'secret',
+                              uri_parameters={'scopeId': 'scopeId'}),
+            ('e', {'name': 'd', 'objectId': 'e'})
+        )
+
+        # object not exist
+        client_session.read = mock.Mock(return_value={
+            'body': {'a': [{
+                'name': 'b',
+                'objectId': 'c'
+            }, {
+                'name': 'd',
+                'objectId': 'e'
+            }]}, 'status': 204
+        })
+        self.assertEqual(
+            common.nsx_search(client_session, 'body/a', 'g', 'secret',
+                              uri_parameters={'scopeId': 'scopeId'}),
+            (None, None)
+        )
+
+    @pytest.mark.internal
+    @pytest.mark.unit
+    def test_nsx_struct_get_list(self):
```

Comment:
Same as above- one test_method per separate test please.
(roughly speaking- one test_method for every time you are calling the method being tested)